### PR TITLE
Improve mem tests, add plotting, fix compare when tests are missing

### LINKED ISSUE #16: compare() not working
`compare()` results in error.

```
using SystemBenchmark
using DataFrames
using CSV

r1 = DataFrame(CSV.File("result1.txt"))
r2 = DataFrame(CSV.File("result2.txt"))

compare(r1, r2)

ERROR: BoundsError: attempt to access 0-element Array{UInt64,1} at index [1]
Stacktrace:
 [1] getindex at ./array.jl:788 [inlined]
 [2] getindex(::WeakRefStrings.StringArray{Union{Missing, String},1}, ::Int64) at [removed]/.julia/packages/WeakRefStrings/lqf5B/src/WeakRefStrings.jl:258
 [3] compare(::DataFrame, ::DataFrame) at [removed]/.julia/packages/SystemBenchmark/JMO7G/src/SystemBenchmark.jl:76
 [4] top-level scope at REPL[11]:1
```

```
(@v1.4) pkg> st
Status `~/.julia/environments/v1.4/Project.toml`
  [336ed68f] CSV v0.6.2
  [a93c6f00] DataFrames v0.21.0
  [30cdaa97] SystemBenchmark v0.2.1 #master (https://github.com/ianshmean/SystemBenchmark.jl)
```

```
julia> versioninfo()
Julia Version 1.4.1
Commit 381693d3df* (2020-04-14 17:20 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2630 v3 @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, haswell)
```
However, `comparetoref()` runs fine.

### PR DESCRIPTION
**Pull request by IanButterworth**
Adds:
- More granular  and lower-level memory copy tests that are designed to capture L1/2/3 and CPU <> RAM performance
- Fix `compare()` when fields are missing in the first resultset (fixes #16, replaces PR #17)
- Add some basic plotting, but not as part of package, given Plots.jl doesn't work on all platforms (Definitely welcome help on the plotting & reporting)
image

### CODE REVIEW

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: CompatHelper
+
+on:
+  schedule:
+    - cron: '00 00 * * *'
```
Comment (giordano) [inline .github/workflows/CompatHelper.yml]: This is a personal taste choice, but I tend to randomise the `cron` time, just not to spawn gazillions of jobs at midnight UTC.  I usually do this:
```julia
julia> "$(rand(0:59)) $(rand(0:23)) * * *"
"38 0 * * *"
```

Comment (IanButterworth) [reply .github/workflows/CompatHelper.yml]: Adopted, thanks

File: reporting/plotting.jl [hunk 2]
```diff
@@ -0,0 +1,85 @@
+using SystemBenchmark
+using CSV
+using DataFrames
+using Plots
+using Statistics
+gr()
+
+function plotreport(df,scale=2.0)
+    
+    platforms = ["Windows", "macOS", "Linux (x86", "Linux (aarch"]
```
Comment (giordano) [inline reporting/plotting.jl:10]: Any reason why they are called `Linux (x86` and `Linux (aarch`?

Comment (IanButterworth) [reply reporting/plotting.jl:10]: It's me being lazy catching the difference in Linux strings in the OS field. Maybe the quick fix would be to remove the '(' chars after used as a filter, and before they're used as labels

Comment (IanButterworth) [reply reporting/plotting.jl:10]: Fixed

**Comment by IanButterworth** [comment]
The core tests are passing, but the anonymous github api request in `getsubmittedbenchmarks` is being rate limited due to all the concurrent CI. That might need some thought if it keeps happening.

I'll merge and release

**Pull request merged**